source/codegen/templates/service.cpp.mako

```diff
@@ -12,6 +12,8 @@ if len(config["custom_types"]) > 0:
   custom_types = config["custom_types"]
 (input_custom_types, output_custom_types) = common_helpers.get_input_and_output_custom_types(functions)
 has_async_functions = any(service_helpers.get_async_functions(functions))
+function_names = service_helpers.filter_proto_rpc_functions_to_generate(functions)
+any_non_mockable_functions = any([not common_helpers.can_mock_function(functions[name]['parameters']) for name in function_names])
 %>\
```

Comment:
I think this needs a comment. You could also potentially grab the library name here to link that requirement (though I'm not sure that's all that clear?).